src/core/xref.js

```diff
@@ -384,23 +422,23 @@ function disambiguate(fetchedData, context, term) {
         valid = [...validTypes].some(type => type === entry.type);
       }
     }
-    if (valid && contextFor) {
+    if (valid && contextFor && entry.for) {
       valid = entry.for.includes(contextFor);
     }
     return valid;
   });
 
   if (!data.length) {
     // no match found
-    return { error: "NO_MATCH", term };
+    throw { error: "NO_MATCH", term };
   }
 
   if (data.length === 1) {
     return data[0]; // unambiguous
   }
 
   // ambiguous
-  return { error: "AMBIGUOUS", term, specs: data.map(e => e.spec) };
+  throw { error: "AMBIGUOUS", term, specs: data.map(e => e.spec) };
```

Comment:
Chatted with @sidvishnoi about making this into real `Error`  by making a new error class.